### PR TITLE
CDAP-2133: Upgrade tool deletes stream conversion adapters

### PR DESCRIPTION
**Pull request by rohitsinha54**
Does the necessary work needed to upgrade Adapters from 2.8 and 3.0 by removing stream-conversion adapter.
There are two places from where the entries need to be removed. 

1. DatasetBasedTimeScheduleStore: Stores all the schedules (triggers) and jobs. 
2. Store (App MetaData Store): Stores the meta data of the adapter.

So, we delete these entries for the stream-conversion adapter across all namespaces.

Build: http://builds.cask.co/browse/CDAP-DT20-1

### CODE REVIEW

File: cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java [hunk 1]
```diff
@@ -505,4 +594,52 @@ private FileMetaDataManager getFileMetaDataManager() {
...
+   *
+   * @return {@link DatasetBasedTimeScheduleStore}
+   * @throws SchedulerException
+   */
+  private DatasetBasedTimeScheduleStore getDatasetBasedTimeScheduleStore() throws SchedulerException {
+    if (datasetBasedTimeScheduleStore == null) {
+      datasetBasedTimeScheduleStore = injector.getInstance(Key.get(DatasetBasedTimeScheduleStore.class,
+                                                                   Names.named("datasetBasedTimeScheduleStore")));
+      // need to call initialize on datasetBasedTimeScheduleStore
+      ExecutorThreadPool threadPool = new ExecutorThreadPool(1);
```
Comment (rohitsinha54) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java]: This is to create and initialize Quartz scheduler which is needed to initialize the DatasetBasedTimeScheduleStore. 
This is present AppFabricServiceRuntimeModule#getScheduler which we use in injection but is not exposed and is used to get the Supplier. I talked to @albertshau and we were not able to find out a way to get/reuse that one.
@chtyim : Any ideas ? Thanks.

File: cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java [hunk 2]
```diff
@@ -423,6 +481,37 @@ private void performUpgrade() throws Exception {
...
+   *
+   * @throws SchedulerException
+   */
+  private void upgradeAdapters() throws SchedulerException {
+    DatasetBasedTimeScheduleStore datasetBasedTimeScheduleStore = getDatasetBasedTimeScheduleStore();
+    NamespaceAdmin namespaceAdmin = injector.getInstance(NamespaceAdmin.class);
+    List<NamespaceMeta> namespaceMetas = namespaceAdmin.listNamespaces();
+    for (NamespaceMeta namespaceMeta : namespaceMetas) {
+      Collection<AdapterSpecification> adapters = getAdapterService().getAdapters(Id.Namespace
+                                                                                    .from(namespaceMeta.getName()));
```
Comment (albertshau) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java]: nit: namespaceMeta.getName() is used several places, would be less verbose to store it in a variable.

File: cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java [hunk 3]
```diff
@@ -505,4 +594,52 @@ private FileMetaDataManager getFileMetaDataManager() {
     }
     return fileMetaDataManager;
   }
+
+  /**
+   * gets the {@link AdapterService}
+   *
+   * @return {@link AdapterService}
+   */
+  private AdapterService getAdapterService() {
```
Comment (albertshau) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java]: dont need this method, just set it after the injector is setup.

File: cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java [hunk 4]
```diff
@@ -216,6 +247,32 @@ public LogSaverTableUtil getLogSaverTableUtil(DatasetFramework dsFramework,
 
         @Provides
         @Singleton
+        @Named("scheduleStoreTableUtil")
```
Comment (albertshau) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java]: how come there are all these provider methods? Shouldn't things be bound already in the modules already in use?

**Comment by albertshau** [comment]
lgtm, though lets open a jira to clean up the guice setup in this tool...

**Comment by rohitsinha54** [comment]
Build passed here: http://builds.cask.co/browse/CDAP-DT20-5

**Pull request merged**